lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Role/Relationships.pm

```diff
@@ -11,11 +11,16 @@ around serialize => sub
     my ($orig, $self, $entity, $inc, $opts, $toplevel) = @_;
     my $ret = $self->$orig($entity, $inc, $opts, $toplevel);
 
-    return $ret unless defined $inc && $inc->has_rels;
+    return $ret unless
+        defined $inc &&
+        $inc->has_rels &&
+        $entity->are_relationships_loaded;
 
     my @rels = map { serialize_entity($_, $inc, $opts) }
-        sort_by { $_->target_key . $_->link->type->name }
-            @{ $entity->relationships };
+        sort_by { $_->link->type->name }
+        sort_by { $_->link_order // 0 }
+        sort_by { $_->target_key }
```

Comment:
@gentlecat Yeah, they're just functions without the parentheses so it reads RTL. `sort_by({ $_ }, sort_by({ $_ },  sort_by({ $_ }, ...)))`.

@uklauer Can you explain? My understanding is that relationships with equal link type names will then be sorted by link_order (if the sort is stable).